diff -up weechat-0.4.3/src/plugins/aspell/weechat-aspell.c.orig weechat-0.4.3/src/plugins/aspell/weechat-aspell.c
--- weechat-0.4.3/src/plugins/aspell/weechat-aspell.c.orig	2014-04-09 11:51:10.699477733 -0500
+++ weechat-0.4.3/src/plugins/aspell/weechat-aspell.c	2014-04-09 11:42:24.537253180 -0500
@@ -990,12 +990,13 @@ weechat_aspell_buffer_closed_cb (void *d
 /*
  * Display infos about external libraries used.
  */
-
+/*
 int
 weechat_aspell_debug_libs_cb (void *data, const char *signal,
                               const char *type_data, void *signal_data)
 {
     /* make C compiler happy */
+/*
     (void) data;
     (void) signal;
     (void) type_data;
@@ -1012,9 +1013,9 @@ weechat_aspell_debug_libs_cb (void *data
     weechat_printf (NULL, "  %s: aspell (?)", ASPELL_PLUGIN_NAME);
 #endif
 #endif /* USE_ENCHANT */
-
+/*
     return WEECHAT_RC_OK;
-}
+}*/
 
 /*
  * Initializes aspell plugin.
@@ -1066,8 +1067,8 @@ weechat_plugin_init (struct t_weechat_pl
                          &weechat_aspell_window_switch_cb, NULL);
     weechat_hook_signal ("buffer_closed",
                          &weechat_aspell_buffer_closed_cb, NULL);
-    weechat_hook_signal ("debug_libs",
-                         &weechat_aspell_debug_libs_cb, NULL);
+    /*weechat_hook_signal ("debug_libs",
+                         &weechat_aspell_debug_libs_cb, NULL);*/
 
     return WEECHAT_RC_OK;
 }
